adlfs/multithread.py

```diff
@@ -135,7 +135,7 @@ def touch(self, src, dst):
             with open(dst, 'wb'):
                 pass
 
-        self.client.run(nthreads, monitor, before_scatter=touch)
+        self.client.run(nthreads, monitor, before_start=touch)
```

Comment:
`invalidate_cache` is not called upon finalization.

We may have to hold off on the file existence check. I talked with Ben about the conditions for forcing an overwrite. I should create a separate issue to track this.